app/src/main/java/com/sdp13epfl2021/projmag/MainActivity.kt

```
@@ -36,12 +39,36 @@ class MainActivity : AppCompatActivity() {
         finish()
     }
 
-    fun goToProjectsList() {
+    fun goToList(addExtras: (Intent) -> Unit) {
         val intent = Intent(this, ProjectsListActivity::class.java)
+        addExtras(intent)
         startActivity(intent)
         finish()
     }
 
+    fun handleLink() {
+        FirebaseDynamicLinks.getInstance().getDynamicLink(intent).addOnSuccessListener {
+                pendingDynamicLinkData ->
+            var dynamicLink: Uri? = null
+            if (pendingDynamicLinkData != null) {
+                dynamicLink = pendingDynamicLinkData.link!!
```

should really throw an exception ?

---

No I would say it should just continue to open the app.